File: arena-backend/src/entities/message.entity.ts
import {
  Column,
  CreateDateColumn,
  DeleteDateColumn,
  Entity,
  Index,
  JoinColumn,
  ManyToOne,
  PrimaryColumn,
  UpdateDateColumn,
} from "typeorm";
import { ChannelEntity } from "./channel.entity";
import { UserEntity } from "./user.entity";

@Entity({ name: "messages" })
@Index("idx_messages_channel_seq", ["channelId", "seq"])
export class MessageEntity {
  @PrimaryColumn({ type: "text" })
  messageId: string;

  @Column({ type: "text" })
  channelId: string;

  @Column({ type: "text" })
  senderId: string;

  @Column({ type: "bigint" })
  seq: number;

  @Column({ type: "text" })
  content: string;

  @CreateDateColumn({ type: "timestamptz" })
  createdAt: Date;

  @UpdateDateColumn({ type: "timestamptz" })
  updatedAt: Date;

  @DeleteDateColumn({ type: "timestamptz", nullable: true })
  deletedAt: Date | null;

  @ManyToOne(() => ChannelEntity)
  @JoinColumn({ name: "channelId", referencedColumnName: "channelId" })
  channel: ChannelEntity;

  @ManyToOne(() => UserEntity)
  @JoinColumn({ name: "senderId", referencedColumnName: "userId" })
  sender: UserEntity;
}

File: legacy/backend/src/main/java/app/sandori/arena/api/domain/message/MessageEntity.java
package app.sandori.arena.api.domain.message;

import app.sandori.arena.api.domain.user.UserEntity;
import app.sandori.arena.api.global.entity.BaseTimeEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.Id;
import jakarta.persistence.Index;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "messages", indexes = {
        @Index(name = "idx_messages_channel_seq", columnList = "channelId, seq")
})
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class MessageEntity extends BaseTimeEntity {

    @Id
    private String messageId;

    @Column(nullable = false)
    private String channelId;

    @Column(nullable = false)
    private String senderId;

    @Column(nullable = false)
    private Long seq;

    @Column(nullable = false, columnDefinition = "text")
    private String content;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "senderId", referencedColumnName = "userId", insertable = false, updatable = false)
    private UserEntity sender;
}
